src/main/java/com/programmerdan/minecraft/banstick/handler/BanStickEventHandler.java

```diff
@@ -320,6 +320,14 @@ public void run() {
 							List<BSIPData> proxyChecks = BSIPData.allByIP(bsPlayer.getLatestSession().getIP());
 							if (proxyChecks != null) {
 								for (BSIPData proxyCheck : proxyChecks) {
+									//check if entire provider is banned
+									if (BanStick.getPlugin().getRegistrarHandler().isBanned(proxyCheck)) {
+										BanHandler.doUUIDBan(player.getUniqueId(), true);
```

Comment:
I suppose a UUID ban is OK, although wouldn't a proxy ban be better? or a new ban type, for Registrar Ban? That way you could unban them all at once. As this is written, you'd have to manually unban each impacted player.

Thoughts?

Reply:
What's the difference between a UUID ban and proxy ban?

So far the use case has only been "I want to unban this single guy who logged on with his VPN" and never wanting to unban everyone with that registrar, so that kinda is how I'd want it to work. Could see though how this might not be ideal for other use cases.

Reply:
Hmm, that makes sense. I'm not functionally opposed to this, just occurred to me if you changed your mind en-masse there was no way to undo the mass-bans, since they were individually done.

One question though; if you unban that player but they use the VPN again, this will simply ban them again. Is that intended? I know for other ban types I have specific exemptions to prevent this -- where you unban + add exemption to prevent cyclic ban situations.

Reply:
If I'm understanding this code right, line 311 should make proxy exclusions also prevent these kind of bans, which seemed to make sense to me and was what I intended. Is that the case?